elastic/kibana
# [Fleet] Add showOnlyActiveDatastream to /packages/installed endpoint

## Summary

Resolve https://github.com/elastic/kibana/issues/191990

Add a new option to installed package endpoint to only return active datastream. 
the query params is a little different than in the original issue `showOnlyActiveDataStreams` as it seems more explict to me (feel free to suggest a better name)


When passing that new params the API will only return existing datastreams pattern.


## Performance concerns

I tested this will all the package installed, and 200 datastreams and it seems there is no performance issue (tested locally)

## Manual tests

You can install some package and verify that the datastreams are not in the installed pacakge response unless you add some data.

```
http GET http://localhost:5601/api/fleet/epm/packages/installed\?showOnlyActiveDataStreams\=true kbn-xsrf:xx elastic-api-version:2023-10-31   --auth=elastic:changeme
```
## Todo 

- [x] Add API integration tests
